### PR TITLE
[app_dart] Fix sublist index when dividing builders for buildbucket query

### PR DESCRIPTION
**Pull request by keyonghan**
This fixes: https://github.com/flutter/flutter/issues/76889

### CODE REVIEW

File: app_dart/lib/src/service/luci.dart [hunk 1]
```diff
@@ -112,7 +112,7 @@ class LuciService {
     const RetryOptions r = RetryOptions(maxAttempts: 3);
     for (int j = 0; j < builders.length; j += _buildersBatchSize) {
       final List<LuciBuilder> partialBuilders =
-          builders.sublist(j, min(j + _buildersBatchSize - 1, builders.length - 1));
+          builders.sublist(j, min(j + _buildersBatchSize, builders.length));
```
Comment (CaseyHillers) [inline app_dart/lib/src/service/luci.dart:115]: Can we add a regression test?

Comment (keyonghan) [reply app_dart/lib/src/service/luci.dart]: Added.

File: app_dart/lib/src/service/luci.dart [hunk 2]
```diff
@@ -99,6 +99,17 @@ class LuciService {
     return results;
   }
 
+  @visibleForTesting
+
```
Comment (CaseyHillers) [inline app_dart/lib/src/service/luci.dart:103]: nit: newline

File: app_dart/test/service/luci_test.dart [hunk 3]
```diff
@@ -102,4 +102,34 @@ void main() {
     final List<Build> resultBuilds = await service.getBuildsForBuilderList(<LuciBuilder>[builder], repo: 'flutter');
     expect(resultBuilds, builds);
   });
+
+  test('luci getPartialBuildersList works correctly', () async {
```
Comment (CaseyHillers) [inline app_dart/test/service/luci_test.dart:106]: Would it be clearer to say `getPartialBuildersList handles non-uniform batches`?

**Review by CaseyHillers — APPROVED** [review]
LGTM w/ comments - Thanks for the quick fix!

File: app_dart/lib/src/service/luci.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/lib/src/service/luci.dart:103]: Do you meant to remove this newline? New line is auto generated via format.
Moved the doc above the annotation.

File: app_dart/test/service/luci_test.dart (hunk 3 above)
Comment (keyonghan) [reply app_dart/test/service/luci_test.dart:106]: Updated.

**Pull request merged**